File: app/src/main/java/cn/ifafu/ifafu/data/local/converter/IntListConverter.java
package cn.ifafu.ifafu.data.local.converter;

import com.alibaba.fastjson.JSONObject;

import org.greenrobot.greendao.converter.PropertyConverter;

import java.util.List;

public class IntListConverter implements PropertyConverter<List<Integer>, String> {

    @Override
    public List<Integer> convertToEntityProperty(String databaseValue) {
        return JSONObject.parseArray(databaseValue, Integer.TYPE);
    }

    @Override
    public String convertToDatabaseValue(List<Integer> entityProperty) {
        return JSONObject.toJSONString(entityProperty);
    }
}

File: app/src/main/java/cn/ifafu/ifafu/data/db/converter/IntListConverter.kt
package cn.ifafu.ifafu.data.db.converter

import androidx.room.TypeConverter
import com.alibaba.fastjson.JSONObject

class IntListConverter {
    @TypeConverter
    fun convertToEntityProperty(databaseValue: String): List<Int> {
        return JSONObject.parseArray(databaseValue, Integer.TYPE)
    }

    @TypeConverter
    fun convertToDatabaseValue(entityProperty: List<Int>): String {
        return JSONObject.toJSONString(entityProperty)
    }

}